main/src/cgeo/geocaching/utils/Formatter.java

```diff
@@ -324,5 +324,33 @@ public static String formatBytes(final long bytes) {
         return String.format(Locale.getDefault(), "%.1f %sB", bytes / Math.pow(1024, exp), pre);
     }
 
+    public static List<CharSequence> truncateCommonSubdir(@NonNull final List<CharSequence> directories) {
+        if (directories.size() < 2) {
+            return directories;
+        }
+        final String firstDir = directories.get(0).toString();
+        String commonEnding = null;
+        int offset = 0;
+        LOOP:
+        while (true) {
```

Comment:
I'm not happy either. Ideas are welcome. 
I've written a Unit test for it, so refactorings can be verified easily. 
Thanks for your effort.